lib/_http_client.js

```diff
@@ -477,70 +477,79 @@ function socketOnEnd() {
   socket.destroy();
 }
 
-function socketOnData(d) {
+function socketOnReadable() {
   const socket = this;
   const req = this._httpMessage;
   const parser = this.parser;
 
   assert(parser && parser.socket === socket);
 
-  const ret = parser.execute(d);
-  if (ret instanceof Error) {
-    prepareError(ret, parser, d);
-    debug('parse error', ret);
-    freeParser(parser, req, socket);
-    socket.destroy();
-    req.socket._hadError = true;
-    req.emit('error', ret);
-  } else if (parser.incoming && parser.incoming.upgrade) {
-    // Upgrade (if status code 101) or CONNECT
-    const bytesParsed = ret;
-    const res = parser.incoming;
-    req.res = res;
-
-    socket.removeListener('data', socketOnData);
-    socket.removeListener('end', socketOnEnd);
-    socket.removeListener('drain', ondrain);
-
-    if (req.timeoutCb) socket.removeListener('timeout', req.timeoutCb);
-    socket.removeListener('timeout', responseOnTimeout);
+  while (true) {
```

Comment:
I would suggest avoiding this always true condition in terms of memory leak issues, and looking at enough complexity of the logic below we probably cannot be sure of triggering the `break` in some cases.